File: LottieSample/src/main/java/com/airbnb/lottie/samples/QRScanActivity.java
package com.airbnb.lottie.samples;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.graphics.PointF;
import android.os.Vibrator;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.dlazaro66.qrcodereaderview.QRCodeReaderView;

import butterknife.BindView;
import butterknife.ButterKnife;

public class QRScanActivity extends AppCompatActivity
    implements QRCodeReaderView.OnQRCodeReadListener {

  @BindView(R.id.qrdecoderview) QRCodeReaderView qrCodeReaderView;
  private Vibrator vibrator;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_qrscan);
    ButterKnife.bind(this);

    vibrator = (Vibrator) getSystemService(Context.VIBRATOR_SERVICE);

    /*
     * Initialize the QR Scanner component
     */
    qrCodeReaderView.setQRDecodingEnabled(true);
    qrCodeReaderView.setAutofocusInterval(2000L);
    qrCodeReaderView.setBackCamera();
    qrCodeReaderView.setOnQRCodeReadListener(this);
    qrCodeReaderView.setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View v) {
        qrCodeReaderView.forceAutoFocus();
      }
    });

  }

  @Override
  protected void onPause() {
    super.onPause();
    qrCodeReaderView.stopCamera();
  }

  @Override public void onQRCodeRead(String s, PointF[] pointFS) {
    vibrator.vibrate(100);

    Intent resultIntent = new Intent();
    resultIntent.putExtra(AnimationFragment.EXTRA_URL, s);
    setResult(Activity.RESULT_OK, resultIntent);
    finish();
  }

  @Override
  protected void onResume() {
    super.onResume();
    qrCodeReaderView.startCamera();
  }



}

File: LottieSample/src/main/kotlin/com/airbnb/lottie/samples/QRScanActivity.kt
package com.airbnb.lottie.samples

import android.content.Context
import android.content.Intent
import android.graphics.PointF
import android.os.Bundle
import android.os.Vibrator
import androidx.appcompat.app.AppCompatActivity
import com.airbnb.lottie.samples.databinding.QrscanActivityBinding
import com.airbnb.lottie.samples.model.CompositionArgs
import com.airbnb.lottie.samples.utils.vibrateCompat
import com.airbnb.lottie.samples.utils.viewBinding
import com.dlazaro66.qrcodereaderview.QRCodeReaderView

class QRScanActivity : AppCompatActivity(), QRCodeReaderView.OnQRCodeReadListener {
    private val binding: QrscanActivityBinding by viewBinding()
    private val vibrator by lazy { getSystemService(Context.VIBRATOR_SERVICE) as Vibrator }

    // Sometimes the qr code is read twice in rapid succession. This prevents it from being read
    // multiple times.
    private var hasReadQrCode = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding.qrView.setQRDecodingEnabled(true)
        binding.qrView.setAutofocusInterval(2000L)
        binding.qrView.setBackCamera()
        binding.qrView.setOnQRCodeReadListener(this)
        binding.qrView.setOnClickListener { binding.qrView.forceAutoFocus() }
    }

    override fun onResume() {
        super.onResume()
        binding.qrView.startCamera()
        hasReadQrCode = false
    }

    override fun onPause() {
        super.onPause()
        binding.qrView.stopCamera()
    }

    override fun onQRCodeRead(url: String, pointFS: Array<PointF>) {
        if (hasReadQrCode) return
        hasReadQrCode = true
        vibrator.vibrateCompat(100)
        finish()
        startActivity(PlayerActivity.intent(this, CompositionArgs(url = url)))
    }

    companion object {
        fun intent(context: Context) = Intent(context, QRScanActivity::class.java)
    }
}